File: myfs/lib/models/student.dart
/// Học sinh – khớp entity Student bên backend.
class Student {
  final int id;
  final String studentCode;
  final String fullName;
  final String? dateOfBirth;
  final String? gender;
  final int? classId;
  final int? parentId;
  final String? className;

  const Student({
    required this.id,
    required this.studentCode,
    required this.fullName,
    this.dateOfBirth,
    this.gender,
    this.classId,
    this.parentId,
    this.className,
  });

  factory Student.fromJson(Map<String, dynamic> json) => Student(
    id: json['id'] as int,
    studentCode: json['studentCode'] as String,
    fullName: json['fullName'] as String,
    dateOfBirth: json['dateOfBirth'] as String?,
    gender: json['gender'] as String?,
    classId: json['classId'] as int?,
    parentId: json['parentId'] as int?,
    className: json['className'] as String?,
  );
}

File: backend/src/main/java/com/myfs/backend/model/Student.java
package com.myfs.backend.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import java.time.LocalDate;

@Getter @Setter
@Entity @Table(name = "student")
public class Student {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY) private Integer id;
    @Column(name = "student_code", nullable = false, unique = true, length = 20) private String studentCode;
    @Column(name = "full_name", nullable = false, length = 100) private String fullName;
    @Column(name = "date_of_birth") private LocalDate dateOfBirth;
    @Column(length = 10) private String gender;
    @Column(name = "class_id") private Integer classId;
    @Column(name = "parent_id") private Integer parentId;
    @Column(name = "student_account_id", unique = true) private Integer studentAccountId;
}
